### PR TITLE
Sp 500 display active plan objectives

### PR DESCRIPTION
**Pull request by Coupey**

### CODE REVIEW

File: app/activeplan/homepage/get.controller.js [hunk 1]
```diff
@@ -1,26 +1,51 @@
...
   let meetings
+  let comments
+  let objectives
 
   delete req.session.planStarted
 
   // get the sentence plan
   try {
-    sentencePlan = await getSentencePlan(planId, token)
+    ;({ comments, objectives } = await getSentencePlan(planId, token))
```
Comment (jonhindle) [inline app/activeplan/homepage/get.controller.js]: remove semi colon and brackets

File: app/activeplan/homepage/get.controller.js [hunk 2]
```diff
@@ -1,26 +1,51 @@
...
   }
 
+  // determine the status of each objective
+  objectives = objectives.map(objective => {
+    const currentObjective = objective
+
+    // objectives default to active if not caught with the rules below
+    currentObjective.type = 'active'
+
+    if (currentObjective.actions.every(action => ['NOT_STARTED'].includes(action.status))) {
```
Comment (jonhindle) [inline app/activeplan/homepage/get.controller.js]: destruct: {status}

File: app/activeplan/homepage/get.controller.js (hunk 1 above)
Comment (jonhindle) [reply app/activeplan/homepage/get.controller.js]: They're defined elsewhere 👍 
Sorry, my mistake 😬

File: app/activeplan/homepage/get.controller.js [hunk 3]
```diff
@@ -1,26 +1,51 @@
...
+      currentObjective.type = 'future'
+    } else if (
+      currentObjective.actions.every(action =>
+        ['COMPLETED', 'PARTIALLY_COMPLETED', 'ABANDONED'].includes(action.status)
+      )
```
Comment (jonhindle) [inline app/activeplan/homepage/get.controller.js]: Not for this, but I think we need to create a some status constants and utils 👍 
Its referenced in multiple places across the app.

File: app/activeplan/homepage/get.controller.js [hunk 4]
```diff
@@ -1,26 +1,51 @@
...
+    ) {
+      currentObjective.type = 'closed'
+    }
+    return currentObjective
+  })
+
+  objectives = groupBy(objectives, item => {
+    return item.type
+  })
```
Comment (jonhindle) [inline app/activeplan/homepage/get.controller.js]: `objectives = groupBy(objectives, ({ type }) => type)`

File: app/activeplan/homepage/get.controller.js (hunk 3 above)
Comment (Coupey) [reply app/activeplan/homepage/get.controller.js]: agreed

**Pull request merged**